src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,211 @@
+            rightSpeedController = new SpeedControllerGroup(frontRightSpark, rearRightSpark);
+
+            frontLeftEncoder = frontLeftSpark.getEncoder();
+            frontRightEncoder = frontRightSpark.getEncoder();
+            rearLeftEncoder = rearLeftSpark.getEncoder();
+            rearRightEncoder = rearRightSpark.getEncoder();
+
+            robotDrive = new DifferentialDrive(leftSpeedController, rightSpeedController);
+            frontLeftPositionController = new SparkPositionController(frontLeftSpark, smartMotionSettings);
+            frontRightPositionController = new SparkPositionController(frontRightSpark, smartMotionSettings);
+            rearLeftPositionController = new SparkPositionController(rearLeftSpark, smartMotionSettings);
+            rearRightPositionController = new SparkPositionController(rearRightSpark, smartMotionSettings);
+
+            autoController = new PositionDriveController(frontRightSpark, rearRightSpark,
+                    frontLeftSpark, rearLeftSpark, smartMotionSettings, positionBuffer,
+                    new EncoderInchesConverter(1 / RobotConstants.DIMENSIONS.MOTOR_REVOLUTIONS_PER_INCH));
+
+    }
+    
+    public void setSpeedMode() {
+        if (drive) {
+            speedSettings.configureSparkMax(frontLeftSpark);
+            speedSettings.configureSparkMax(frontRightSpark);
+            speedSettings.configureSparkMax(rearLeftSpark);
+            speedSettings.configureSparkMax(rearRightSpark);
+        }
+    }
+
+        public void setPositionMode() {
+        if (drive) {
+            smartMotionSettings.configureSparkMax(frontLeftSpark);
+            smartMotionSettings.configureSparkMax(frontRightSpark);
+            smartMotionSettings.configureSparkMax(rearLeftSpark);
+            smartMotionSettings.configureSparkMax(rearRightSpark);
+        }
+    }
+    
+    public EncoderValues getEncoderValues() {
+        if (drive) {
+            return new EncoderValues(frontLeftEncoder.getPosition(),
+                    rearLeftEncoder.getPosition(),
+                    frontRightEncoder.getPosition(),
+                    rearRightEncoder.getPosition());
+        }
+        return EncoderValues.NO_ENCODER_VALUES;
+    }
+
+    @Override
+    public void customPeriodic(RobotPose rp, FieldPose fp) {
+        if (drive) {
+            logger.log("Front Left Encoder Ticks", frontLeftEncoder.getPosition());
+            logger.log("Front Right Encoder Ticks", frontRightEncoder.getPosition());
+            logger.log("Rear Left Encoder Ticks", rearLeftEncoder.getPosition());
+            logger.log("Rear Right Encoder Ticks", rearRightEncoder.getPosition());
+            logger.log("isSafetyEnabled", robotDrive.isSafetyEnabled());
+            if (getCurrentCommand() != null) {
+                logger.log("current command", getCurrentCommand().getName());
+            }
+        }
+        latestRobotPose = rp;
```

Comment:
Do we need to update the latestRobotPose here?  If possible, I think we need to limit the Periodic routines to simply do logging.